### PR TITLE
🚸 Stop the loading spinner if the product fetch fails

### PR DESCRIPTION
**Pull request by wenche**
Resolve [AB#86853](https://dev.azure.com/EquinorASA/d02c350d-092d-41bf-a812-f05d5d8ad1b0/_workitems/edit/86853)

### CODE REVIEW

File: pages/api/popular.ts [hunk 1]
```diff
@@ -85,7 +95,7 @@ const PopularAssetsHandler: NextApiHandler = async (req, res) => {
       authString,
       approvedStatusResp.body.id,
       dataProductTypeIdRes.body.results[0]?.id,
-      limit,
+      limit
```
[the remark this answers is not in the document]
Comment (wenche) [reply pages/api/popular.ts:98]: Yes, I don't think thats a part of the es5. Would still vote for `trailingComma: 'all'` :)

File: hooks/usePopularProducts.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      setIsLoading(true)
+      try {
+        const { body } = await HttpClient.get("/api/popular", { query: { limit: 6 } })
+        if (!ignore) {
+          setPopularDataProducts(body)
+        }
+        setIsLoading(false)
+      } catch (err) {
+        let msg: string = String(err)
+        if (err instanceof Error) {
```
[the remark this answers is not in the document]
Comment (wenche) [reply hooks/usePopularProducts.ts:26]: What is superfluous? It's how we do it for the other hooks as well, if you remember that we discussed the error handling mechanism in typescript from the kent c dodds article?

[the remark this answers is not in the document]
Comment (aronhoyer) [reply hooks/usePopularProducts.ts:26]: oh right it's to appease the typescript compiler. never mind then

File: pages/api/popular.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aronhoyer) [reply pages/api/popular.ts:98]: and you can change it at any time 😛

**Review by aronhoyer — APPROVED** [review]

File: pages/api/popular.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (wenche) [reply pages/api/popular.ts:98]: Will do. One day.

**Pull request merged**